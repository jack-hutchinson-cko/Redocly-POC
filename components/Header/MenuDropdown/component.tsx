import React from 'react';
import { StyledTopLabel } from './menuDropdown.styles';
import { StyledContainer, StyledItem, StyledText } from './menuDropdown.styles';
import { ArrowRight, ApiRef, FAQ } from '../../svg';

const MenuDropdown = (): JSX.Element => (
	<StyledContainer>
		<StyledItem href="https://docs.checkout.com" target="_blank">
			<StyledTopLabel>
				<ApiRef />
				Guides
			</StyledTopLabel>
			<StyledText>
				Hey, come look at the guides
				<ArrowRight />
			</StyledText>
		</StyledItem>
		<StyledItem href="https://docs.checkout.com/four" target="_blank">
			<StyledTopLabel>
				<FAQ />
				FAQ
			</StyledTopLabel>
			<StyledText>
				Find answers to our most frequently asked questions
				<ArrowRight />
			</StyledText>
		</StyledItem>
	</StyledContainer>
);

export default MenuDropdown;
