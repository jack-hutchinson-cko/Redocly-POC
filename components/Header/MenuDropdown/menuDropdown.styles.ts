import styled from 'styled-components';
import { colours } from '../../css/tokens';

export const StyledContainer = styled.div`
	background: white;
	border: 1px solid ${colours.S05};
	box-shadow: 0px 2px 5px rgba(12, 17, 66, 0.15);
	display: flex;
	flex-direction: column;
	font-family: 'Graphik LCG Web';
	position: relative;
	gap: 16px;
	padding: 32px;
	width: 452px;
	position: absolute;
	top: 64px;
	z-index: 1000;
`;

export const StyledSVG = styled.div`
	display: flex;
	margin: auto;
	transform: ${(props: { isOpen: boolean }) => (props.isOpen ? 'rotate(180deg)' : 'none')};
`;

export const StyledTopLabel = styled.div`
	display: flex;
	gap: 12px;
`;

export const StyledItem = styled.div``;

export const StyledText = styled.p`
	display: flex;
	align-items: center;
	font-size: 14px;
	gap: 8px;
	margin: 8px 0 0;
`;
