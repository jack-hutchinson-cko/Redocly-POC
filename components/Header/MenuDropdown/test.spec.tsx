import { shallow } from 'enzyme';
import React from 'react';
import MenuDropdown from './component';

describe('MenuDropdown component', () => {
	test('component matches snapshot', () => {
		expect.assertions(1);

		const wrapper = shallow(<MenuDropdown />);

		expect(wrapper).toMatchSnapshot();
	});
});
