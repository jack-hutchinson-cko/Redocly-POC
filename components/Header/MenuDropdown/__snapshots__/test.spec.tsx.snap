// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuDropdown component component matches snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <ApiRef />
      Guides
    </styled.div>
    <styled.p>
      Hey, come look at the guides
      <ArrowRight />
    </styled.p>
  </styled.div>
  <styled.div>
    <styled.div>
      <FAQ />
      FAQ
    </styled.div>
    <styled.p>
      Find answers to our most frequently asked questions
      <ArrowRight />
    </styled.p>
  </styled.div>
</styled.div>
`;
